mixins/localize-mixin.js

```diff
@@ -15,30 +15,42 @@ export const LocalizeMixin = superclass => class extends superclass {
 		super();
 
 		this.__documentLocaleSettings = getDocumentLocaleSettings();
-		let first = true;
-		this.__languageChangeCallback = () => {
-
-			if (!this._hasResources()) return;
-			const possibleLanguages = this._generatePossibleLanguages();
-			this.constructor.getLocalizeResources(possibleLanguages)
-				.then((res) => {
-					if (!res) {
-						return;
-					}
-					this.__language = res.language;
-					this.__resources = res.resources;
-					if (first) {
-						first = false;
-					} else {
-						this._languageChange();
-					}
-				});
+		this.__resourcesLoadedPromise = new Promise((resolve) => {
```

Comment:
I know this is sort of separate concern, but do you have any appetite for updating the mixin so that the work only gets done once per instance of a component by either storing it on the class (not instance) or in a weak map outside of the class?

Reply:
I think that's captured in #232?

Reply:
I think so yep - wasn't sure if we wanted to tackle it while refactoring this...?

Reply:
Haha I personally do not, but I can take a crack at it in a follow-up.